connector/avro/src/main/scala/org/apache/spark/sql/avro/AvroDeserializer.scala

```diff
@@ -158,7 +158,7 @@ private[sql] class AvroDeserializer(
         }
       case LONG =>
         (logicalDataType, catalystType) match {
-          case (LongType, LongType) => (updater, ordinal, value) =>
+          case (_, LongType) => (updater, ordinal, value) =>
```

Comment:
so any physical avro long type can be read as spark long type? does it mean we can read a daytime interval in avro as long type?

Reply:
Got it. Let me smaller the scope. Now it's only allow the logical type
```
    "time-millis",
    "time-micros",
    "timestamp-micros",
    "timestamp-millis",
    "local-timestamp-millis",
    "local-timestamp-micros"
```